src/libs/ReportUtils.ts

```diff
@@ -7089,6 +7090,14 @@ function findPolicyExpenseChatByPolicyID(policyID: string): OnyxEntry<Report> {
     return Object.values(ReportConnection.getAllReports() ?? {}).find((report) => isPolicyExpenseChat(report) && report?.policyID === policyID);
 }
 
+function getSourceIDFromReportAction(reportAction: OnyxEntry<ReportAction>): string {
+    const message = Array.isArray(reportAction?.message) ? reportAction?.message?.at(-1) ?? null : reportAction?.message ?? null;
```

Comment:
There's still duplicated code in ContextMenuActions

https://github.com/Expensify/App/blob/3fd16441c2d4606e28f7696fd86943a3525355ce/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L40-L43

https://github.com/Expensify/App/blob/3fd16441c2d4606e28f7696fd86943a3525355ce/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L518-L521

Reply:
@situchan I think using the utils function in the onPress function does not bring much benefit because variables like originalFileName, sourceURL, sourceURLWithAuth, etc., are all reused below.